### PR TITLE
Fix unreliable unit test

### LINKED ISSUE #16018: RecentBlogPostsQueryTests.ShouldListBlogPostWhenCallingAQuery is flaky
### Describe the bug

The `RecentBlogPostsQueryTests.ShouldListBlogPostWhenCallingAQuery` test can fail randomly, without the code it tests being wrong, and then succeed on the nest run. I.e., it's flaky.

### To Reproduce
I assume that if you run it enough times, eventually, you'll see a random fail. This happened during [this build](https://github.com/OrchardCMS/OrchardCore/actions/runs/9003004131/job/24732624592?pr=16015) too:

image

A rerun fixed it. I can't reproduce this locally.

I've seen unit tests being flaky before too, but can't recall any specifics. Please add them here if you notice anything else.

### Expected behavior
The test only fails if the code is incorrect.

### Screenshots
See above.

### PR DESCRIPTION
**Pull request by gvkries**
Fixes #16018.

### CODE REVIEW

File: test/OrchardCore.Tests/Apis/GraphQL/Queries/RecentBlogPostsQueryTests.cs [hunk 1]
```diff
@@ -12,6 +12,14 @@ public async Task ShouldListBlogPostWhenCallingAQuery()
             using var context = new BlogContext();
             await context.InitializeAsync();
 
+            // The RecentBlogPosts query sorts the content items by CreatedUtc. Since
+            // the CreatedUtc property is managed automatically and cannot be set when
+            // adding a content item, adding this delay ensures that the second post is
+            // always newer.
+            // The delay was chosen to be higher than the clock resolution in most
+            // operating systems.
+            await Task.Delay(20);
```
Comment (sebastienros) [inline test/OrchardCore.Tests/Apis/GraphQL/Queries/RecentBlogPostsQueryTests.cs:21]: Let's use `Assert.Collection` to ensure two items with specific strings

Comment (sebastienros) [reply test/OrchardCore.Tests/Apis/GraphQL/Queries/RecentBlogPostsQueryTests.cs:21]: Or, add DocumentId as a second order criteria by default in GraphQL

File: test/OrchardCore.Tests/Apis/GraphQL/Queries/RecentBlogPostsQueryTests.cs [hunk 2]
```diff
@@ -35,11 +35,17 @@ public async Task ShouldListBlogPostWhenCallingAQuery()
...
-            Assert.Equal("Some sorta blogpost in a Query!", nodes[0]["displayText"].ToString());
-            Assert.Equal("Man must explore, and this is exploration at its greatest", nodes[1]["displayText"].ToString());
+            Assert.NotNull(jsonArray);
+            Assert.Equal(2, jsonArray.Count);
+
+            // The RecentBlogPosts query sorts the content items by CreatedUtc. If the
+            // test is executing too fast, both blog entries may have the same CreatedUtc
+            // value and ordering becomes random. Because of this, we do not assert the order
+            // of the result.
+            Assert.Contains("Some sorta blogpost in a Query!", jsonArray.Select(node => node["displayText"]?.ToString()));
```
Comment (MikeAlhayek) [inline test/OrchardCore.Tests/Apis/GraphQL/Queries/RecentBlogPostsQueryTests.cs:47]: Can you save the result of `jsonArray.Select(node => node["displayText"]?.ToString())` into a variable and use it in both lines?

Comment (gvkries) [reply test/OrchardCore.Tests/Apis/GraphQL/Queries/RecentBlogPostsQueryTests.cs]: Sure, done!

**Review by MikeAlhayek — APPROVED** [review]

**Pull request merged**